### PR TITLE
C4 (Plastique) Improvements

### PR DESCRIPTION
**Pull request by Geevies**
* You can now examine C4 planted on walls to see when it detonates, you can also hit it with tools to access its wires.
* C4 is no longer visible from the other side of the wall, instead its true location is where you're standing on the moment you plant it.
* Planting C4 no longer forces you to touch the wall before the slap it down. It also doesn't play a dropping sound the moment you apply it.
* You now only do the attack animation once you place the C4, instead of before the timer starts.

### CODE REVIEW

**Comment by Geevies** [comment]
!review

File: code/game/objects/effects/plastic_explosive.dm [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+		"NORTHEAST" = list(32, 32),
+		"EAST" = list(32, 0),
+		"SOUTHEAST" = list(32, -32),
+		"SOUTH" = list(0, -32),
+		"SOUTHWEST" = list(-32, -32),
+		"WEST" = list(-32, 0),
+		"NORTHWEST" = list(-32, 32)
+	)
+	var/list/pixel_shifts = dir_to_pixel[dir_text]
+	if(islist(pixel_shifts))
```
Comment (mikomyazaki) [inline code/game/objects/effects/plastic_explosive.dm:34]: This will always return true?

File: code/game/objects/effects/plastic_explosive.dm [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+/obj/effect/plastic_explosive/Initialize(var/atom/owner_pos, var/atom/target, var/obj/item/plastique/c4)
+	. = ..()
+	parent = c4
+	parent.effect_overlay = src
+	parent.forceMove(src)
+	name = parent.name
+	desc = parent.desc
+	set_position(get_dir(src, target))
+
+/obj/effect/plastic_explosive/proc/set_position(var/dir)
```
Comment (mikomyazaki) [inline code/game/objects/effects/plastic_explosive.dm:20]: Can this whole thing be done with the DIR2PIXEL_X and DIR2PIXEL_Y macros?

File: code/game/objects/effects/plastic_explosive.dm [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+		"NORTHWEST" = list(-32, 32)
+	)
+	var/list/pixel_shifts = dir_to_pixel[dir_text]
+	if(islist(pixel_shifts))
+		pixel_x = pixel_shifts[1]
+		pixel_y = pixel_shifts[2]
+
+/obj/effect/plastic_explosive/examine(mob/user)
+	. = ..()
+	if(Adjacent(user))
```
Comment (mikomyazaki) [inline code/game/objects/effects/plastic_explosive.dm:40]: Examine can already handle this, if you pass it a distance argument.

e.g. 

. = ..(distance = 0)
if(.)
   to_chat(whatever)

File: code/modules/mining/mine_items.dm [hunk 4]
```diff
@@ -1286,7 +1286,7 @@ var/list/total_extraction_beacons = list()
 		target = get_atom_on_turf(src)
 	if(!target)
 		target = src
-	target.cut_overlay(image_overlay, TRUE)
+	QDEL_NULL(effect_overlay)
```
Comment (mikomyazaki) [inline code/modules/mining/mine_items.dm:1289]: Why not
```suggestion
	cut_overlay(effect_overlay)
```

Comment (Geevies) [reply code/modules/mining/mine_items.dm:1289]: the overlay is in the form of an object, now

File: code/game/objects/effects/plastic_explosive.dm (hunk 2 above)
Comment (Geevies) [reply code/game/objects/effects/plastic_explosive.dm:20]: No, that doesn't handle diagonals

File: code/game/objects/effects/plastic_explosive.dm (hunk 3 above)
Comment (Geevies) [reply code/game/objects/effects/plastic_explosive.dm:40]: image

Comment (Geevies) [reply code/game/objects/effects/plastic_explosive.dm:40]: image

**Comment by Geevies** [comment]
!review

File: code/game/objects/effects/plastic_explosive.dm (hunk 3 above)
Comment (mikomyazaki) [reply code/game/objects/effects/plastic_explosive.dm:39]: You'll need to add it to the possible args for /obj/examine too then.

**Review by Alberyk — APPROVED** [review]

File: code/game/objects/effects/plastic_explosive.dm (hunk 3 above)
Comment (Geevies) [reply code/game/objects/effects/plastic_explosive.dm:39]: this distance thing doesn't account for things like adjacent does, so i won't be using it.

File: code/game/objects/effects/plastic_explosive.dm [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+	parent = c4
+	parent.effect_overlay = src
+	parent.forceMove(src)
+	name = parent.name
+	desc = parent.desc
+	set_position(get_dir(src, target))
+
+/obj/effect/plastic_explosive/proc/set_position(var/dir)
+	var/dir_text = uppertext(dir2text(dir))
+	// DIR = list(pixel_x, pixel_y)
```
Comment (NonQueueingMatt) [inline code/game/objects/effects/plastic_explosive.dm:22]: delete this

File: code/game/objects/effects/plastic_explosive.dm [hunk 6]
```diff
@@ -0,0 +1,46 @@
+/obj/effect/plastic_explosive
+	name = "plastic explosives"
+	desc = "Used to put holes in specific areas without too much extra hole."
+	gender = PLURAL
+	icon = 'icons/obj/assemblies.dmi'
+	icon_state = "plastic-explosive2"
+	anchored = TRUE
+	density = FALSE
+	var/obj/item/plastique/parent
```
Comment (NonQueueingMatt) [inline code/game/objects/effects/plastic_explosive.dm:9]: clean this up in Destroy

**Review by NonQueueingMatt — CHANGES_REQUESTED** [review]

**Review by NonQueueingMatt — APPROVED** [review]

**Pull request merged**